build-system/tasks/visual-diff.js

```diff
@@ -143,10 +139,27 @@ async function launchWebServer() {
     host: HOST,
     port: PORT,
     retries: WEBSERVER_TIMEOUT_RETRIES, // retry timeout defaults to 1 sec
-  }).on('connected', resolver).on('timeout', rejecter);
+  }).on('connected', () => {
+    return resolver(gulpServeAsync);
+  }).on('timeout', rejecter);
   return deferred;
 }
 
+/**
+ * Kill the webserver process and this own process.
+ *
+ * @param {!ChildProcess} webServerProcess the webserver process to shut down.
+ */
+async function shutdown(webServerProcess) {
+  if (webServerProcess.exitCode == null) {
```

Comment:
Is this check meant to see if `webServerProcess` is still running? If so, you could use `webServerProcess.connected`. See [reference](https://nodejs.org/api/child_process.html#child_process_subprocess_connected).